lib/Service/SettingsService.php

```diff
@@ -0,0 +1,328 @@
+			if (!$folder instanceof Folder) {
+				return [];
+			}
+			return $folder->getDirectoryListing();
+		} catch (NotFoundException $e) {
+			return [];
+		}
+	}
+
+	/**
+	 * Generate file URL.
+	 *
+	 * @param string $type
+	 * @param string $category
+	 * @param string $fileName
+	 * @return string
+	 */
+	private function generateFileUri(string $type, string $category, string $fileName, string $userId): string {
+		
+		// Passing userId is dangerous so we have to trim from url...
+		if (strpos($type, '/') !== false) {
+			$type = explode('/', $type)[0];
+		}
+
+		$token = $this->generateIframeToken($type, $userId);
+		
+		return $this->urlGenerator->linkToRouteAbsolute(
+			'richdocuments.settings.getSettingsFile',
+			[
+				'type' => $type,
+				'token' => $token['token'],
+				'category' => $category,
+				'name' => $fileName,
+			]
+		);
+	}
+
+	/**
+	 * Get a specific settings file.
+	 *
+	 * @param string $type
+	 * @param string $category
+	 * @param string $name
+	 * @return ISimpleFile
+	 */
+	public function getSettingsFile(string $type, string $category, string $name): ISimpleFile {
+		try {
+			$baseFolder = $this->appData->getFolder($type);
+		} catch (NotFoundException $e) {
+			throw new NotFoundException("Type folder '{$type}' not found.");
+		}
+		
+		try {
+			$categoryFolder = $baseFolder->getFolder($category);
+		} catch (NotFoundException $e) {
+			throw new NotFoundException("Category folder '{$category}' not found in type '{$type}'.");
+		}
+		
+		try {
+			return $categoryFolder->getFile($name);
+		} catch (NotFoundException $e) {
+			throw new NotFoundException("File '{$name}' not found in category '{$category}' for type '{$type}'.");
+		}
+	}
+
+	/**
+	 * Delete a specific settings file from the type/category directory.
+	 *
+	 * @param string $type
+	 * @param string $category
+	 * @param string $name
+	 */
+	public function deleteSettingsFile(string $type, string $category, string $name, string $userId): void {
+		try {
+			$baseFolder = $this->appData->getFolder($type);
+		} catch (NotFoundException $e) {
+			throw new NotFoundException("Type folder '{$type}' not found.");
+		}
+
+		if ($type === 'userconfig') {
+			try {
+				$baseFolder = $baseFolder->getFolder($userId);
+			} catch (NotFoundException $e) {
+				throw new NotFoundException("User folder '{$userId}' not found.");
+			}
+		}
+		
+		try {
+			$categoryFolder = $baseFolder->getFolder($category);
+		} catch (NotFoundException $e) {
+			throw new NotFoundException("Category folder '{$category}' not found in type '{$type}'.");
+		}
+
+		try {
+			if (!$categoryFolder->fileExists($name)) {
+				throw new NotFoundException("File '{$name}' not found in category '{$category}' for type '{$type}'.");
+			}
+			$categoryFolder->getFile($name)->delete();
+		} catch (NotFoundException $e) {
+			throw $e;
+		} catch (NotPermittedException $e) {
+			throw $e;
+		}
+	}
+
+	// TODO: Handle installDefaultSystemFiles setting
```

Comment:
Please track todos as checklist items in the pr description rather then in the code so we can file them in follow up issues on merge